internal/metrics/reporter_test.go

```diff
@@ -0,0 +1,110 @@
+package metrics_test
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/lightstep/lightstep-tracer-common/golang/gogo/metricspb"
+	"github.com/lightstep/lightstep-tracer-go/internal/metrics"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Reporter", func() {
+	var reporter *metrics.Reporter
+	var ingestRequest metricspb.IngestRequest
+
+	JustBeforeEach(func() {
```

Comment:
General suggestion: avoid `JustBeforeEach` if you can help it. IME, it always leads to more pain than it's worth - confusing control flows, weird race conditions, enabling of overly complicated tests...

Reply:
I'm not too familiar with ginko, does `BeforeEach` also suffer from this?

Reply:
Nope :) `BeforeEach` behaves more consistently with the equivalents from other test harnesses out there. Basically, both run FIFO, but `JustBeforeEach`s run _after_ all `BeforeEach`s... which means that an outer layer's `JustBeforeEach` can override an inner layer's `BeforeEach`.

```go
var _ = Describe("grandparent", func() {
	BeforeEach(func() {
		fmt.Println("grandparent BeforeEach")
	})

	JustBeforeEach(func() {
		fmt.Println("grandparent JustBeforeEach")
	})

	Describe("parent", func() {
		BeforeEach(func() {
			fmt.Println("parent BeforeEach")
		})

		JustBeforeEach(func() {
			fmt.Println("parent JustBeforeEach")
		})

		Describe("child", func() {
			BeforeEach(func() {
				fmt.Println("child BeforeEach")
			})

			JustBeforeEach(func() {
				fmt.Println("child JustBeforeEach")
			})

			It("runs", func() {
				Expect(1 + 1).To(Equal(2))
			})
		})
	})
})
```

prints:
```
grandparent BeforeEach
parent BeforeEach
child BeforeEach
grandparent JustBeforeEach
parent JustBeforeEach
child JustBeforeEach
```